src/components/Video/Controls/Controls.tsx

```diff
@@ -21,11 +21,38 @@ const Controls = ({
   // Handle play pause
   */
   const [isPlaying, setIsPlaying] = useState(isAutoplaying);
+  const [shareLinkText, setShareLinkText] = useState('Share');
+  const [isShareComponent, setIsShareComponent] = useState(false);
```

Comment:
This could be named a bit better, something like `showShareDisalog` or similar.
If you want to take it further, the share button can be its own component, that handles thes state and the copying inside there, to keep this Controls component a bit tidier.